interface: added tool to track unneeded renders

tylershuster opened the pull request:
As per a discussion with @liam-fitzgerald a while back, adds `why-did-you-render` during development to check for unneeded renders. Seems to be pretty clean already, but it looks like it doesn't track functional components by default (only PureComponents), but might aid in future debugging.

Also removes some unnecessary `console.log`s

matildepark commented:
Seems innocent enough, just wondering what this actually looks like in development now.

tylershuster commented:
Give it a shot. Pretty clean. The clock apparently rerenders once due to a strictly equal object. Should be more useful by doing:
```jsx
const FunctionalComponent = (props) => <div></div>;
FunctionalComponent.whyDidYouRender = true;
export default FunctionalComponent;
```

matildepark commented:
I have literally no idea how this changes the experience of developing or where to find this information. I boot up dev environment and it's just Landscape.

tylershuster commented:
It won't change visually. Open the console. There will be one exemplary log message on Launch.

Add the above code to a Functional Component which you believe is rerendering unnecessarily.

matildepark commented:
<img width="790" alt="image" src="https://user-images.githubusercontent.com/20846414/110833014-f182f300-8269-11eb-8f9d-9085df3001e1.png">

?

tylershuster commented:
show the clock

matildepark commented on pkg/interface/package.json:
```diff
@@ -13,6 +13,7 @@
     "@tlon/indigo-react": "^1.2.19",
     "@tlon/sigil-js": "^1.4.3",
     "@urbit/api": "file:../npm/api",
+    "@welldone-software/why-did-you-render": "^6.1.0",
```
This should specifically be a dev dependency. This sort of utility seems like it shouldn't be in the bundle at all, in the same way I don't bundle VSCode with Landscape, but I don't think React has such a concept, so that's moot.

matildepark reviewed (commented):
Fine to add, given the change here — though I note we don't seem to use pure components, I never see this in console in Chat, even though ChatMessage ostensibly *is* one? Maybe it's a perfect render.

tylershuster commented:
Ok, I updated the dev dependency. I added `LinkResource.whyDidYouRender = true` and confirmed that you need to add that to functional components, and also that there are some other rerenders.
